### ISSUE TITLE
[dart2wasm] --print-kernel no longer works after the Wasm builder changes

### ISSUE THREAD

**Issue by osa1** [dart-lang/sdk#53174]
To repro, try `--print-kernel` on any program:

```
Unhandled exception:
Value not yet finalized
#0      Finalizable.value (package:wasm_builder/src/ir/finalizable.dart:22)
#1      Indexable.index (package:wasm_builder/src/ir/indexable.dart:15)
#2      Translator.translate (package:dart2wasm/translator.dart:306)
#3      compileToModule (package:dart2wasm/compile.dart:138)
<asynchronous suspension>
#4      main (package:dart2wasm/dart2wasm.dart:134)
<asynchronous suspension>
#5      main (file:///usr/local/google/home/omersa/dart/sdk/sdk/pkg/dart2wasm/bin/dart2wasm.dart:7)
<asynchronous suspension>
```

cc @joshualitt

**Comment by osa1** [comment]
Fix in https://dart-review.googlesource.com/c/sdk/+/319820.

**Comment by osa1** [comment]
Fixed in fc35778ff719dc67f31b4bf28a05d5e1b3cd722f.
